src/main/java/uk/gov/hmcts/dts/fact/services/admin/list/AdminContactTypeService.java

```diff
@@ -0,0 +1,103 @@
+package uk.gov.hmcts.dts.fact.services.admin.list;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.dts.fact.exception.DuplicatedListItemException;
+import uk.gov.hmcts.dts.fact.exception.ListItemInUseException;
+import uk.gov.hmcts.dts.fact.exception.NotFoundException;
+import uk.gov.hmcts.dts.fact.model.admin.ContactType;
+import uk.gov.hmcts.dts.fact.repositories.ContactRepository;
+import uk.gov.hmcts.dts.fact.repositories.ContactTypeRepository;
+
+import java.util.Comparator;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Service
+@Slf4j
+public class AdminContactTypeService {
+
+    private final ContactTypeRepository contactTypeRepository;
+    private final ContactRepository contactRepository;
+
+    @Autowired
+    public AdminContactTypeService(final ContactTypeRepository contactTypeRepository, final ContactRepository contactRepository) {
+        this.contactTypeRepository = contactTypeRepository;
+        this.contactRepository = contactRepository;
+
+    }
+
+    public List<ContactType> getAllContactTypes() {
+        return contactTypeRepository.findAll()
+            .stream()
+            .map(ContactType::new)
+            .sorted(Comparator.comparing(ContactType::getType))
+            .collect(toList());
+    }
+
+    public ContactType getContactType(final Integer id) {
+        try {
+            return new ContactType(contactTypeRepository.getOne(id));
+        } catch (final javax.persistence.EntityNotFoundException exception) {
+            throw new NotFoundException(exception);
+        }
+    }
+
+    @Transactional
+    public ContactType createContactType(final ContactType contactType) {
+        checkIfContactTypeAlreadyExists(contactType.getType());
+        return new ContactType(contactTypeRepository.save(createNewContactTypeEntityFromModel(contactType)));
+    }
+
+    @Transactional
+    public ContactType updateContactType(final ContactType updatedContactType) {
```

Comment:
Just to double check, is the logic in place here to prevent us from updating an existing type to another via the name? 

I think we are only updating descriptions for now, so shouldn't be a problem right?

Reply:
The logic to prevent us from updating an existing type to another is currently in the original api pull request https://github.com/hmcts/fact-api/pull/453 as the fact-admin functional test were dependent on it. Once that pull request is merged in I will merge the master branch to this branch before merging.

Reply:
Wonderful, thanks!